### PR TITLE
Migrate MQTT broker to ActiveMQ

### PR DESCRIPTION
**Pull request by richturner**

### CODE REVIEW

File: manager/src/main/java/org/openremote/manager/mqtt/DefaultMQTTHandler.java [hunk 1]
```diff
@@ -102,65 +77,33 @@ public void start(Container container) throws Exception {
...
-
-                if (canPublish(connection, lastWillTopic)) {
-                    MqttFixedHeader fixedHeader = new MqttFixedHeader(MqttMessageType.PUBLISH, true, MqttQoS.EXACTLY_ONCE, false, 0);
-                    MqttPublishVariableHeader varHeader = new MqttPublishVariableHeader(lastWill.getTopic(), 0);
-                    ByteBuf payload = Unpooled.wrappedBuffer(lastWill.getPayload().toString().getBytes());
-                    InterceptPublishMessage message = new InterceptPublishMessage(new MqttPublishMessage(fixedHeader, varHeader, payload), msg.getClientID(), msg.getUsername());
-                    doPublish(connection, lastWillTopic, message);
-                }
-            }
-        }
```
Comment (MartinaeyNL) [inline manager/src/main/java/org/openremote/manager/mqtt/DefaultMQTTHandler.java:163]: Last will functionality removed? Wasn't this implemented recently?

Comment (MartinaeyNL) [reply manager/src/main/java/org/openremote/manager/mqtt/DefaultMQTTHandler.java:163]: Got clarification; this is automatically handled by ActiveMQ. Tests are still there to cover it.

**Review by MartinaeyNL — COMMENTED** [review]
No major issues found when reading through the code.
It looks really impressive and rock solid to me, especially comparing it to how I would code 😉 

Made a few basic comments, you can change if necessary.
Let me know when ready to merge, so I can approve the PR.

File: manager/src/main/java/org/openremote/manager/provisioning/UserAssetProvisioningMQTTHandler.java [hunk 2]
```diff
@@ -186,17 +184,24 @@ public boolean canSubscribe(MqttConnection connection, Topic topic) {
...
-    public void doSubscribe(MqttConnection connection, Topic topic, InterceptSubscribeMessage msg) {
+    public void onSubscribe(RemotingConnection connection, Topic topic) {
         // Nothing to do here as we'll publish to this topic in response to client messages
     }
 
     @Override
-    public void doUnsubscribe(MqttConnection connection, Topic topic, InterceptUnsubscribeMessage msg) {
+    public void onUnsubscribe(RemotingConnection connection, Topic topic) {
+
+    }
```
[the remark this answers is not in the document]
Comment (richturner) [reply manager/src/main/java/org/openremote/manager/provisioning/UserAssetProvisioningMQTTHandler.java:194]: Yes we can leave as no harm, but also could be valid to remove :-)

File: manager/src/main/java/org/openremote/manager/mqtt/MQTTBrokerService.java [hunk 3]
```diff
@@ -0,0 +1,473 @@
...
+            for (MQTTHandler handler : getCustomHandlers()) {
+                handler.onConnect(remotingConnection);
+            }
+        }
+    }
+
+    @Override
+    public void afterDestroyConnection(RemotingConnection connection) throws ActiveMQException {
+
+    }
```
[the remark this answers is not in the document]
Comment (richturner) [reply manager/src/main/java/org/openremote/manager/mqtt/MQTTBrokerService.java:345]: Yes we can leave as no harm, but also could be valid to remove :-)

**Review by MartinaeyNL — APPROVED** [review]

**Pull request merged**